Doctrine/Orm/Filter/BooleanFilter.php

```diff
@@ -22,7 +22,7 @@ class BooleanFilter extends BaseBooleanFilter
 
     public function __construct(
         ManagerRegistry $managerRegistry,
-        ?RequestStack $requestStack = null,
+        RequestStack $requestStack = null,
```

Comment:
This is odd. What makes an argument optional is the `= null` part. I guess this works because a non-nullable type-hinted argument which defaults to null is actually _required_: Passing a null value will result in an exception. 

I would remove all _default null_ stuff but keep the nullable '?RequestStack'. Since you can't omit it nor pass null to them, it is a safe change.

Reply:
Removing the null from all parameters like this 

```
    public function __construct(
        ManagerRegistry $managerRegistry,
        ?RequestStack $requestStack,
        LoggerInterface $logger,
        array $properties,
        NameConverterInterface $nameConverter,
        ResourceMetadataFactoryInterface $resourceMetadataFactory
    ) {
```
results in:

```
Ivoz\Api\Doctrine\Orm\Filter\BooleanFilter::__construct(): Argument #4 ($properties) must be of type array, null given, called in /opt/symfony/var/cache/dev/Container7qZPzEy/getIvoz_Api_Filter_BooleanService.php on line 33
```

Reply:
Also, it may be a reason, but the rest of the objects outside Doctrine/Orm/Filter that have that constructor parameter have neither null nor ?  (p.e. DataAccessControlParser, CsvMixer or EntityDenormalizer)

Reply:
I had to review the documentation. It’s a subtlety that affects versions between 8.1 and 8.4.

From php.net:
```php
function foo($a = [], $b) {}     // Default not used; deprecated as of PHP 8.0.0
function foo($a, $b) {}          // Functionally equivalent, no deprecation notice

function bar(A $a = null, $b) {} // As of PHP 8.1.0, $a is implicitly required
                                 // (because it comes before the required one),
                                 // but implicitly nullable (deprecated as of PHP 8.4.0),
                                 // because the default parameter value is null
function bar(?A $a, $b) {}       // Recommended
```

We could make all arguments that are implicitly nullable explicitly nullable (?), just like you have already done with  RequestStack.

Reply:
Decision: We stick this PR to removing the deprecating warning related to optional parameter position and create a new one updating default nullable parameters format